### PR TITLE
[WIP] Add ArviZ installation guide [docs]

### PR DESCRIPTION
**Pull request by arungrace88**
## Description

This is an interim update on issue 'Write installation guide #1487' for your perusal. The .rst file contains the guide to install ArviZ using either pip or conda forge. The required and optional dependencies are also discussed in the guide. Kindly have a look and let me know the feedback. Thanks.

### CODE REVIEW

**Review by OriolAbril — COMMENTED** [review]
This is great, thank you for the contribution. I have added several comments, to try and start some discussion and back and forth feedback, they are not must do changes. 

I also wanted to note a couple extra points. First is that I would place the document inside the getting started folder, and for the file to actually be shown in docs, it has to be added at the toctree in https://github.com/arviz-devs/arviz/blob/main/doc/source/getting_started/index.md (I would put it in the first place).

Second extra comment is that I think that one can install _all_ optional dependencies with `pip install arviz[all]` thanks to https://github.com/arviz-devs/arviz/blob/main/setup.py#L57, can you check and if it works add that to the guide? Not sure if in pip section or if in dependencies section though

I have also added `[docs]` to the PR title to have Azure generate the docs so we have a preview. They can be download from Azure as artifacts.

File: doc/source/installation.rst
Comment (arungrace88) [reply doc/source/installation.rst:10]: Hi @OriolAbril, 
Thanks for the feedback. Very helpful. Agree with the change.

File: doc/source/installation.rst
Comment (arungrace88) [reply doc/source/installation.rst:49]: I was able to generate the documentation locally. I will look into other comment.

Comment (arungrace88) [reply doc/source/installation.rst:49]: I had a look at ways to include the .txt file from the parent directory. Seems like '..literalinclude' is a better option than '..include' in our context.

Comment (OriolAbril) [reply doc/source/installation.rst:49]: :ok_hand: thanks for looking into that

File: doc/source/getting_started/Installation.rst
Comment (OriolAbril) [inline doc/source/getting_started/Installation.rst:9]: I would put this line a couple lines later, inside the stable section

File: doc/source/getting_started/Installation.rst
Comment (OriolAbril) [inline doc/source/getting_started/Installation.rst:56]: I would remove this.

**Comment by arungrace88** [comment]
@OriolAbril,
I have updated the Installation.rst file and index.md file in the getting started folder. Double checked  the command ``pip install arviz[all]``, and it installs all the optional dependencies. This is documented in the installation page. Kindly have a look. Thanks.

File: doc/source/getting_started/Installation.rst
Comment (OriolAbril) [inline doc/source/getting_started/Installation.rst:91]:
```suggestion
  If available, ArviZ makes use of faster ujson when :func:`arviz.from_json` is
```

Writing the function like this will add a hyperlink to the corresponding page in the api reference

File: doc/source/getting_started/Installation.rst
Comment (OriolAbril) [inline doc/source/getting_started/Installation.rst:2]:
```suggestion
Installation guide
```

**Review by OriolAbril — APPROVED** [review]
Thanks! It looks great

Before merging, I would add a link to this page in the home page. It currently has pip, conda and dev installation instrucctions, I think it would be great to leave pip and conda and change the dev install with something like "See the installation guide for more details"

File: doc/source/getting_started/Installation.rst
Comment (arungrace88) [reply doc/source/getting_started/Installation.rst:91]: Awesome. Thanks for letting me know.

File: doc/source/getting_started/Installation.rst
Comment (arungrace88) [reply doc/source/getting_started/Installation.rst:2]: Okay

**Comment by arungrace88** [comment]
Hi @OriolAbril, Added changes to index and installation file. Kindly have a look. Thanks.

**Review by OriolAbril — APPROVED** [review]

**Comment by OriolAbril** [comment]
This looks great, thanks @arungrace88 ! I will merge it once it is added to the changelog.

**Comment by OriolAbril** [comment]
I took the liberty of updating the changelog myself and merged, thanks for the PR @arungrace88

**Pull request merged**
